pkg/controller/jobs/job/job_controller_test.go

```diff
@@ -313,110 +313,189 @@ var (
+					PodSets(*utiltesting.MakePodSet(kueue.DefaultPodSetName, 10).SetMinimumCount(5).Request(corev1.ResourceCPU, "1").Obj()).
 					Admit(utiltesting.MakeAdmission("cq").AssignmentPodCount(8).Obj()).
 					Obj(),
 			},
-			wantJob: *utiltestingjob.MakeJob("job", "ns").
-				SetAnnotation(JobMinParallelismAnnotation, "5").
-				Suspend(false).
-				Parallelism(8).
-				Request(corev1.ResourceCPU, "1").
-				Image("", nil).
-				Obj(),
 			wantWorkloads: []kueue.Workload{
 				*utiltesting.MakeWorkload("a", "ns").
-					PodSets(*utiltesting.MakePodSet("main", 10).SetMinimumCount(5).Request(corev1.ResourceCPU, "1").Obj()).
+					PodSets(
+						*utiltesting.MakePodSet(kueue.DefaultPodSetName, 10).
+							SetMinimumCount(5).
+							Request(corev1.ResourceCPU, "1").
+							Obj(),
+					).
 					Admit(utiltesting.MakeAdmission("cq").AssignmentPodCount(8).Obj()).
 					Obj(),
 			},
 		},
 		"unsuspended job with partial admission and non-matching admitted workload is suspended and workload is deleted": {
-			job: *utiltestingjob.MakeJob("job", "ns").
+			reconcilerOptions: []jobframework.Option{
+				jobframework.WithManageJobsWithoutQueueName(true),
+			},
+			job: *(&utiltestingjob.JobWrapper{Job: *baseJob.DeepCopy()}).
 				SetAnnotation(JobMinParallelismAnnotation, "5").
 				Suspend(false).
-				Parallelism(10).
-				Request(corev1.ResourceCPU, "1").
-				Image("", nil).
+				Obj(),
+			wantJob: *(&utiltestingjob.JobWrapper{Job: *baseJob.DeepCopy()}).
+				SetAnnotation(JobMinParallelismAnnotation, "5").
 				Obj(),
 			workloads: []kueue.Workload{
 				*utiltesting.MakeWorkload("a", "ns").
-					PodSets(*utiltesting.MakePodSet("main", 10).SetMinimumCount(5).Request(corev1.ResourceCPU, "1").Obj()).
+					PodSets(
+						*utiltesting.MakePodSet(kueue.DefaultPodSetName, 10).
+							SetMinimumCount(5).
+							Request(corev1.ResourceCPU, "1").
+							Obj(),
+					).
 					Admit(utiltesting.MakeAdmission("cq").AssignmentPodCount(8).Obj()).
 					Obj(),
 			},
 			wantErr: jobframework.ErrNoMatchingWorkloads,
-			wantJob: *utiltestingjob.MakeJob("job", "ns").
+		},
+		"the workload is created when queue name is set": {
+			job: *(&utiltestingjob.JobWrapper{Job: *baseJob.DeepCopy()}).
+				Suspend(false).
+				Queue("test-queue").
+				Obj(),
+			wantJob: *(&utiltestingjob.JobWrapper{Job: *baseJob.DeepCopy()}).
+				Queue("test-queue").
+				Obj(),
+			wantWorkloads: []kueue.Workload{
+				*utiltesting.MakeWorkload("job", "ns").
+					PodSets(*utiltesting.MakePodSet(kueue.DefaultPodSetName, 10).Request(corev1.ResourceCPU, "1").Obj()).
+					Queue("test-queue").
+					Priority(0).
+					Obj(),
+			},
+		},
+		"the workload is not created when queue name is not set": {
+			job: *(&utiltestingjob.JobWrapper{Job: *baseJob.DeepCopy()}).
+				Suspend(false).
+				Parallelism(1).
+				Obj(),
+			wantJob: *(&utiltestingjob.JobWrapper{Job: *baseJob.DeepCopy()}).
 				SetAnnotation(JobMinParallelismAnnotation, "5").
```

Comment:
There shouldn't be this annotation.

This also means that our `cmpopts` are too permissive. Instead of ignore the entirety of `ObjectMeta`, we can omit certain fields on it.